site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -328,11 +328,17 @@ public function modifyCourseMaterialsFileTimeStamp($newdatatime): JsonResponse {
     }
 
     private function recursiveEditFolder(array $course_materials, CourseMaterial $main_course_material) {
+        $main_path =  $main_course_material->getPath();
+
         foreach ($course_materials as $course_material) {
-            if (
-                str_starts_with(pathinfo($course_material->getPath(), PATHINFO_DIRNAME), $main_course_material->getPath())
-                && $course_material->getPath() != $main_course_material->getPath()
-            ) {
+            $course_material_path = $course_material->getPath();
+            $course_material_dir = pathinfo($course_material->getPath(), PATHINFO_DIRNAME);
+
+            $same_start = str_starts_with($course_material_dir, $main_path);
+            $not_same_file = $course_material_path != $main_path;
```

Comment:
It's better to use the strict equality operators (`===` and `!==`) in PHP to prevent unexpected type conversions from occurring.

See here for more info: https://www.php.net/manual/en/types.comparisons.php

```suggestion
            $not_same_file = $course_material_path !== $main_path;
```